code/HUXt.py

```diff
@@ -1403,9 +1540,26 @@ def solve_upwind(model, v_input):
     :param v_input: Time series of inner boundary solar wind speeds, in km/s.
     :return:
     """
+    
+    #check the CFL condition is met
+    constants= huxt_constants()
+    alpha = constants['alpha']  # Scale parameter for residual SW acceleration
+    r_accel = constants['r_accel']  # Spatial scale parameter for residual SW acceleration
+    v_max = constants['v_max']
+    
+    #get the radial grid parameters
+    rrel=model.rrel
+    dtdr = model.dtdr
+    Nr=model.Nr
+    
+    #check the CFL condition is met
+    if (np.nanmax(v_input)>v_max):
+        print('Warning: CFL condition not met, time step too short. Increase HUXt.constants.v_max')
```

Comment:
This is a really good point - I'm not sure I did this check anywhere in HUXt1D or HUXt2D. There might be some other points it would be good to add this in - like checking the ConeCME parameters and stuff.